### ISSUE TITLE
Wildcard path issue on Windows

### ISSUE THREAD

**Issue by SID800** [Kintyre/ksconf#59]
New user to this helpful tool... on Windows.   Attempting to run some of the commands, is not working as outlined in the manual.  User error or bug? 

```
C:\>ksconf check c:\splunk\etc\system\default\*.conf
Skipping missing file:  c:\splunk\etc\system\default\*.conf
Completed checking 1 files.  rc=0 Breakdown:
   0 files were parsed successfully.
   0 files failed.

C:\>dir c:\splunk\etc\system\default\*.conf
 Volume in drive C is Windows
 Volume Serial Number is 8ED6-089A

 Directory of c:\splunk\etc\system\default

09/04/2019  11:31             7,489 alert_actions.conf
09/04/2019  11:31             2,737 app.conf
09/04/2019  11:31               450 audit.conf
09/04/2019  11:31             1,376 authentication.conf
09/04/2019  11:31            10,351 authorize.conf
09/04/2019  11:31               781 collections.conf
<snip>

C:\>ksconf --version
 _                         __
| | _____  ___ ___  _ __  / _|
| |/ / __|/ __/ _ \| '_ \| |_
|   <\__ \ (_| (_) | | | |  _|
|_|\_\___/\___\___/|_| |_|_|

ksconf 0.7.6  (Build 435)
Python: 2.7.16  (c:\python27\python.exe)
Git SHA1 cf0cd07d committed on 2019-08-15
Installed at: c:\python27\lib\site-packages\ksconf
Written by Lowell Alleman <lowell@kintyre.co>.
Copyright (c) 2019 Kintyre Solutions, Inc, all rights reserved.
Licensed under Apache Public License v2

  kintyre_splunk_conf  (0.7.6)

    Commands:
      check           (stable)    OK
      combine         (beta)      OK
      diff            (stable)    OK
      filter          (alpha)     OK
      merge           (stable)    OK
      minimize        (beta)      OK
      promote         (beta)      OK
      rest-export     (beta)      OK
      rest-publish    (alpha)     Missing 3rd party module:  No module named splunklib.client
      snapshot        (alpha)     OK
      sort            (stable)    OK
      unarchive       (beta)      OK
      xml-format      (alpha)     OK   (lxml 4.3.4)

```

**Comment by lowell80** [comment]
Thanks for taking the time to write up a bug report!

I see what's happening.  This really isn't a bug or user mistake.  (How often does that happen?)  This is actually a platform issue.  There's a fundamental difference between a proper Unix shell and CMD on Windows.  In shell (like bash, sh... I think even PowerShell get's this right, but I'm not 100% sure), wildcards (like `*` or `?`) are expanded before the program (`ksconf` in this case) is executed.  However, on Windows, wildcarded arguments are passed in as-is.  Therefore ksconf reports `missing file:  c:\splunk\etc\system\default\*.conf`

So while this really isn't technically a *bug* in ksconf, it is really inconvenient and I suspect that most users (including myself) would just expect to be protected from silly OS-level technicality.

  BTW, most testing has been done on Linux/Mac, or from the GIT bash prompt when running on Windows.  But I really want it to run smoothly on all platforms supported by Splunk, so this feedback is very helpful.  Thanks again.


**TL DR**:   This behavior should work transparently.   I'm working on a patch now.

**Comment by lowell80** [comment]
I've just released `0.7.7rc1`.  You can upgrade to it by running:

`pip install -U kintyre-splunk-conf=0.7.7rc1`

I haven't been able to test it fully (as this is a pre-release).  Let me know if this resolve the issue.

**Comment by SID800** [comment]
Thanks!  the RC works  (p.s. had to use pip install -U kintyre-splunk-conf==0.7.7rc1 ) 

```
C:\>ksconf check c:\splunk\etc\system\default\*.conf
Successfully parsed c:\splunk\etc\system\default\alert_actions.conf
Successfully parsed c:\splunk\etc\system\default\app.conf
Successfully parsed c:\splunk\etc\system\default\audit.conf
Successfully parsed c:\splunk\etc\system\default\authentication.conf
Successfully parsed c:\splunk\etc\system\default\authorize.conf
Successfully parsed c:\splunk\etc\system\default\collections.conf
<snip>
Successfully parsed c:\splunk\etc\system\default\workload_rules.conf
Completed checking 47 files.  rc=20 Breakdown:
   43 files were parsed successfully.
   4 files failed.
```

**Comment by lowell80** [comment]
Ah yes, I forgot the `==`  :-)  Good catch!

Glad it's working.  I want to do some more testing and get a unit test in-place to prevent any regressions on this.  I'll get this officially released later on.

Let me know if you run into any more issues.

*This ticket will automatically close when I roll the code changes to the `master` branch.*

**Comment by lowell80** [comment]
Closing this as 0.7.7 has been released.  Not sure why this didn't close automatically.  If anyone has further issues please re-open.
